### DISCUSSION TITLE
You still a failure

### DISCUSSION THREAD

**Discussion by Strypper** [dotnet/maui#18075] [Show and tell]
MORE QUALITY?????

Our team has developed an application to assess MAUI features across different versions and share our development experiences with the community. Our primary goal is to track Microsoft's progress in fixing bugs related to MAUI native, with no focus on MAUI hybrid. If you're interested in Blazor's status within MAUI, this post won't provide that information. The following bugs are based on our app, which does not connect to external APIs; it solely displays built-in controls for testing. We've encountered some frustrating issues:

**Application Status:**
- Version: net7

**Windows Bugs:**
1. **CollectionView:**
   - a. Header template has no effect on the header template.
   - b. In the ITemplate, we display text in HTML mode with bold and blue color enabled, but only the bold formatting works.
   - c. SelectionChanged does not update the binding properties (TwoWay binding is already enabled).
2. **Shell Search Bar:** The app fails to display the search bar.
3. **Button:**
   - a. No hover animation.
   - b. No click animation.
   - c. In some cases, the hit target does not cover the entire button (tested on Android, where the button covers the entire surface).
4. **Frame:**
   - a. Does not resize when items inside increase or decrease in height.
   - b. The theme is not compatible with the Windows platform.
5. **Label:**
   - Line break mode doesn't work.
   - a. MiddleTruncation.
   - b. HeadTruncation.
   - c. No text selection is supported.
6. **Tabbed Page:** Exhibiting unusual behavior.
7. **Slider:** Unable to control the thumb image size.
8. **ToggleSwitch:** Minimum Width - every switch will have a minimum width of 100.
9. **TableView:**
   - a. Table section title texts are forcibly lowercase.
   - b. Hover layer does not cover the entire child UI for Entry Cell.
10. **Image Button:** Applying visual states causes the app to break in release mode.
11. **Carousel:** Child elements behave oddly when the application window width changes.
12. **IndicatorView:**
   - a. Unable to trigger initially - [Link to Issue](https://github.com/dotnet/maui/issues/17727).
   - b. Not positioned correctly when used with a carousel.
13. **Frame Navigation:** Extremely slow and causes memory leaks.
14. **Popup:** DisplayActionSheet popup lacks corner radius.

**Android:**
1. **Grid:** Vertical Option Bottom does not take effect.
2. Other issues not yet tested.

**iOS:**
- Collection View inside the frame overflows.
- Other issues not yet tested.

**After Upgrading to net8rc2:**
- "nope" means the issue is not fixed.

1. **CollectionView:**
   - a. Header template has no effect on the header template - sometimes it works, sometimes it doesn't.
   - b. In the ITemplate, we display text in HTML mode with bold and blue color enabled, but only the bold formatting works - "nope."
   - c. SelectionChanged does not update the binding properties (TwoWay binding is already enabled) - not tested.
2. **Shell Search Bar:** The app still doesn't show the search bar - "nope."
3. **Button:**
   - a. No hover animation - "nope."
   - b. No click animation - "nope."
   - c. In some cases, the hit target doesn't cover the entire button - "nope."
4. **Frame:**
   - a. Now resizes when items inside increase or decrease in height - fixed.
   - b. The theme's compatibility with the Windows platform is not yet tested.
5. **Label:**
   - Line break mode doesn't work.
   - a. MiddleTruncation - "nope."
   - b. HeadTruncation - "nope."
   - c. No text selection is supported - "nope."
6. **Tabbed Page:** Still exhibiting weird behavior - "nope."
7. **Slider:** The issue of not controlling the thumb image size remains - "nope."
8. **ToggleSwitch:** Minimum Width is still set at 100 - "nope."
9. **TableView:**
   - a. Table section title texts are still forcibly lowercase - "nope."
   - b. Hover layer still doesn't cover the entire child UI for Entry Cell - "nope."
10. **Image Button:** Not tested.
11. **Carousel:** There's been a slight improvement, with items trying to snap into view, but issues remain when scrolling with a mouse - somewhat fixed.
12. **IndicatorView:**
   - a. Unable to trigger initially - [Link to Issue](https://github.com/dotnet/maui/issues/17727) - not tested.
   - b. Positioning issues when used with a carousel - somewhat fixed.
13. **Frame Navigation:** Still extremely slow and causing memory leaks - "nope."
14. **Popup:** DisplayActionSheet popup still lacks corner radius - "nope."

**Android:**
1. App crashes when trying to launch, with no error information provided - only this useless piece of info "[libc] Fatal signal 11 (SIGSEGV), code 1 (SEGV_MAPERR), fault addr 0x717bce14b1ef in tid 13001 (yname.mauisland), pid 13001 (yname.mauisland)".
2. For some reason, Firebase uninstalls, and we have to reinstall it.

**iOS:**
1. App crashes at startup.
2. Visual Studio for Mac refuses to understand net8, even though the workload is installed.

**Summary:**
The process of upgrading from version 7 to 8 has been quite painful. It required delving into .csproj files and comparing them with newly created net8 apps to change the version. However, upon completion, we encountered issues where Android and iOS stopped working. Moreover, less than 30% of the problems on the Windows platform have been resolved. This experience has been more trouble than an enhancement in quality.

**Comment by jonmdev** [comment]
It's a real challenge. Every day I encounter new bugs and I am spending half my days trying to figure out where they are coming from, make repro projects to capture them, and post them. I don't know what is going on at Microsoft. This is supposed to be one of the richest software companies on earth? Very hard to get anything done at all. I am going crazy.

**Comment by jsuarezruiz** [comment]
Do you have or know the links to associated issues?. Especially from the crashes mentioned.
Some of the problems have PRs created and I hope that in the next version (GA) they will be resolved. For example:
- https://github.com/dotnet/maui/pull/17821
- https://github.com/dotnet/maui/pull/14918

**Reply by Strypper** [reply]
MAUIsland 
Branch: net8_migration

**Comment by Xyncgas** [comment]
Why did microsoft ever bought xamarin if they are deprecating it today

.NET MAUI.Blazor is great because I use it without using Xamarin side of it but I still get a cross platform app that people can install and display the web UI

Let is instead : 

Generic UI Platform: The idea is to develop a universal UI platform that goes beyond the limitations of HTML and CSS. This is motivated by the need for a more efficient and adaptable solution for cross-platform applications.

Efficiency Over HTML/CSS: HTML and CSS have limitations in terms of performance and flexibility. The proposed platform aims to overcome these limitations and be more efficient, improving the overall user experience.

Binary Format and Interpreter: To achieve this efficiency, a binary format and interpreter would be used, allowing for faster rendering of UI elements. This decision is based on the desire for a streamlined and standardized approach to UI rendering.

Cross-Platform UI Description: The goal is to enable cross-platform applications to describe their UI using this platform. This would streamline development, reduce compatibility issues, and improve cross-device consistency.

.NET Language Interop: Supporting various .NET languages is essential for broad developer adoption. It ensures that developers can work with their preferred language, increasing the platform's appeal.

Unified UI Experience: Providing a consistent UI experience across all devices and applications is crucial for user satisfaction. This idea stems from the desire to offer a seamless experience irrespective of the user's choice of device.

Remote Scripting of UI: Enabling remote scripting of the UI is a response to the growing demand for cloud-native applications. This capability would allow for remote UI control, which is especially relevant in cloud-native and distributed applications.

Efficient Cloud-Native Applications: The proposal suggests that components of cloud-native applications can run more efficiently on the server, reducing the load on the client device. This reasoning is grounded in the need to optimize cloud-native application performance.

C#/F#/VB for UI: By using C#, F#, VB, or other .NET languages, the aim is to create a more native and efficient approach to UI design. This is based on the premise that working in these languages can lead to improved performance and consistency.

Binary UI Description: Using a binary format to describe UI elements is seen as a way to transmit UI information efficiently, like sending video frames. The intention is to provide a high-performance method for UI rendering.

Embedding Multimedia: The proposal suggests embedding images and audio directly into the UI, similar to the web but using C#. This approach aims to simplify multimedia integration and enhance user experiences.

Familiar UI Manipulation: Making UI manipulation similar to HTML/CSS is intended to ease the transition for developers familiar with web technologies, ensuring a more intuitive and user-friendly experience.

Absolute Positioning: Absolute positioning for UI elements (and similar CSS positioning approach) is recommended to simplify actions like implementing right-click menus. This choice is aimed at improving ease of use and developer productivity.

Leadership for Microsoft: The proposal suggests that Microsoft could take a leading role in this initiative, harnessing the potential of a new UI platform to gain a competitive advantage in the technology landscape.

Replacing HTML/CSS: The long-term vision is to replace HTML and CSS, leveraging the proposed platform's performance benefits. The motivation here is to offer a superior alternative for web and cloud-native applications, ultimately enhancing user experiences.

**Comment by Strypper** [comment]
Comfirm on net 8 
Frame navigation improves a lot
